City-of-Helsinki/drupal-hdbt
# UHF-468 announcement theme

# Announcement theme
https://helsinkisolutionoffice.atlassian.net/browse/UHF-468

![image](https://user-images.githubusercontent.com/1191667/134491009-af6313b8-0ed7-4e03-9528-ef5a0aa35ee5.png)
![image](https://user-images.githubusercontent.com/1191667/134491054-cae27dd3-3dce-48ac-bed9-b13f5551bb83.png)


## What was done

* Created a TWIG template for styling announcements under `templates/content/node--announcement--default.html.twig`
* Created a new SCSS file under `src/scss/components/misc/_announcement.scss` to style announcements
* Added two new mixins to end of `src/scss/library/mixins/_mixins.scss`
  * `icon-mask-image`
  * `pseudo-icon`
* Discussed with designer, ending up with some fine tuning to original specs for backend and frontend tasks.

## How to test (existing install)

* If https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/155 has not yet been merged, follow its instructions to get a working version of the notifications on your system.
* On project root run `composer require drupal/hdbt:dev-UHF-468_announcement_theme` 
* `make drush-cr`
* In [Content > Add Content > Annoucement](https://helfi.docker.so/fi/node/add/announcement) add new notifications of all three types with varying text lengths and styles, some with optional link, some without.
* See that styles match the design (note the agreed changes in [this comment](https://helsinkisolutionoffice.atlassian.net/browse/UHF-468?focusedCommentId=46865) )
  * Remember to check mobile and desktop sizes
  * Check the scss, twig codes, their locations and generic coding style (My first bigger commit on project 🙈 )

Also check this PR: City-of-Helsinki/drupal-helfi-platform-config#155
